dev/devicelab/lib/framework/devices.dart

```diff
@@ -330,6 +334,38 @@ class AndroidDeviceDiscovery implements DeviceDiscovery {
     return results;
   }
 
+  /// Collect the device state for specified deviceId.
+  Future<String?> getDeviceState({required String deviceId}) async {
```

Comment:
Do we need to get the full state or sending a single command and collecting the exit code will be enough? e.g. `adb shell ping -C 1`

Reply:
Well I just wanted to make sure that the device was actually ready. And since we can allow a few devices we should target a particular one.